op-service/testutils/mock_rollup_client.go

```diff
@@ -0,0 +1,67 @@
+package testutils
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockRollupClient struct {
+	mock.Mock
+}
+
+func (m *MockRollupClient) OutputAtBlock(ctx context.Context, blockNum uint64) (*eth.OutputResponse, error) {
+	out := m.Mock.MethodCalled("OutputAtBlock", blockNum)
+	return out[0].(*eth.OutputResponse), *out[1].(*error)
+}
+
+func (m *MockRollupClient) ExpectOutputAtBlock(blockNum uint64, response *eth.OutputResponse, err error) {
+	m.Mock.On("OutputAtBlock", blockNum).Once().Return(response, &err)
+}
+
+func (m *MockRollupClient) SyncStatus(ctx context.Context) (*eth.SyncStatus, error) {
+	out := m.Mock.MethodCalled("SyncStatus")
+	return out[0].(*eth.SyncStatus), *out[1].(*error)
+}
+
+func (m *MockRollupClient) ExpectSyncStatus(status *eth.SyncStatus, err error) {
+	m.Mock.On("SyncStatus").Once().Return(status, &err)
+}
+
+func (m *MockRollupClient) RollupConfig(ctx context.Context) (*rollup.Config, error) {
+	out := m.Mock.MethodCalled("RollupConfig")
+	return out[0].(*rollup.Config), *out[1].(*error)
+}
+
+func (m *MockRollupClient) ExpectRollupConfig(config *rollup.Config, err error) {
+	m.Mock.On("RollupConfig").Once().Return(config, &err)
+}
+
+func (m *MockRollupClient) StartSequencer(ctx context.Context, unsafeHead common.Hash) error {
+	out := m.Mock.MethodCalled("StartSequencer", unsafeHead)
+	return *out[0].(*error)
+}
+
+func (m *MockRollupClient) ExpectStartSequencer(unsafeHead common.Hash, err error) {
+	m.Mock.On("StartSequencer", unsafeHead).Once().Return(&err)
+}
+
+func (m *MockRollupClient) SequencerActive(ctx context.Context) (bool, error) {
+	out := m.Mock.MethodCalled("SequencerActive")
+	return out[0].(bool), *out[1].(*error)
```

Comment:
```suggestion
	out := m.Mock.Called()
	return out.Bool(0), out.Error(1)
```
* `mock` uses reflection to get the method name, should work here
* In general, all arguments are passed to `Called`, like here `Called(ctx)`. I can see how in this instance, we never care about the `ctx` value, so might as well skip like you do already 👍🏻 
* `mock` provides convenience return methods for bools, ints, errors etc
* I think it should just be possible to use normal `error`s, without pointers. Then you also have to change the `ExpectSequencerActive` definition to just return `active, err`.

Similarly in all other mock definitions

Reply:
Cool. I might take some time after this PR is done, then, to fix up the other `mock_`s too, I was over-matching their style